tests/by-util/test_pmap.rs

```diff
@@ -211,14 +268,18 @@ fn assert_cmdline_only(pid: &str, s: &str) {
 // ...
 //  total          1040320K
 #[cfg(target_os = "linux")]
-fn assert_format(pid: u32, s: &str) {
+fn assert_format(pid: u32, s: &str, show_path: bool) {
     let (first_line, rest) = s.split_once('\n').unwrap();
     let re = Regex::new(&format!("^{pid}:   .+[^ ]$")).unwrap();
     assert!(re.is_match(first_line));
 
     let rest = rest.trim_end();
     let (memory_map, last_line) = rest.rsplit_once('\n').unwrap();
-    let re = Regex::new(r"(?m)^[0-9a-f]{16} +[1-9][0-9]*K (-|r)(-|w)(-|x)(-|s)- (  \[ (anon|stack) \]|[a-zA-Z0-9._-]+)$").unwrap();
+    let re = if show_path {
+        Regex::new(r"(?m)^[0-9a-f]{16} +[1-9][0-9]*K (-|r)(-|w)(-|x)(-|s)- (  \[ (anon|stack) \]|/[/a-zA-Z0-9._-]+)$").unwrap()
+    } else {
+        Regex::new(r"(?m)^[0-9a-f]{16} +[1-9][0-9]*K (-|r)(-|w)(-|x)(-|s)- (  \[ (anon|stack) \]|[a-zA-Z0-9._-]+)$").unwrap()
+    };
```

Comment:
Maybe it would make sense to split the regexes to make it easier to see the (small) differences? Something like:
```rust
let base_pattern = "<pattern for the identical part>";
let mapping_pattern = if show_path {
    "<pattern path expected>"
} else {
    "<pattern no path expected>"
}

let re = Regex::new(&format("{base_pattern}{mapping_pattern}")).unwrap();
```